core/src/ids.rs

```diff
@@ -98,17 +98,26 @@ macro_rules! declare_id {
 
         impl core::fmt::Display for $name {
             fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-                let mut end = self.0.len();
+                let len = self.0.len();
+                let mut e1 = len / 2;
+                let mut s2 = len / 2;
 
                 if let Some(precision) = f.precision() {
```

Comment:
I'd make these calculation only if precision is good.
```suggestion
                if matches!(f.precision(), Some(precision) if precision < len/2) {
```

Reply:
For me, this works worse than previous variant: it contains for 1 more comparison and 1 more expect that we should proof
```rust
if matches!(f.precision(), Some(precision) if precision < median) {
    e1 = f.precision().expect("Infallible. Checked above.");
    s2 = len - e1;
}
```

Check to be more than zero in meaningless

Reply:
Ok, we can add an internal comparison:

```suggestion
                if let Some(precision) = f.precision() {
                    if precision > median {
```

Zero `precision` looks weird for me.

Reply:
> Ok, we can add an internal comparison:
> 
> Zero `precision` looks weird for me.

But it works for floats:
`assert_eq!(format!("{:.0?}", 5.23), "5")`
The same I see reasonable for ids.
Imagine case when you interested only in payload in logs, for example, and debugging messages sent:
you can go with `log::debug!("{message:.0?}")` to ignore ids

Reply:
Agree.